library/coverage/supported_spec.rb

```diff
@@ -0,0 +1,19 @@
+require_relative '../../spec_helper'
+require 'coverage'
+
+describe "Coverage.supported?" do
+  ruby_version_is "3.2" do
+    it "returns true if coverage measurement is supported for the given mode" do
+      Coverage.supported?(:lines).should == true
+      Coverage.supported?(:branches).should == true
+      Coverage.supported?(:methods).should == true
+      Coverage.supported?(:eval).should == true
+    end
+
+    it "return false if coverage measurement is not supported for the given mode" do
+      Coverage.supported?(:all).should == false
+      Coverage.supported?(:foo).should == false
+      Coverage.supported?(:bar).should == false
+    end
+  end
```

Comment:
It's kind of feature discovering mechanism so it doesn't make sense to require in specs some specific result (true or false).

IMHO the only thing that could be checked is that this method:
- returns `true` or `false` (for existing modes and arbitrary Symbols)
- converts its argument to Symbol

Reply:
TBH I haven't checked. I meant handling of not Symbol arguments in general.